doc/calculator/Cargo.toml

```diff
@@ -6,14 +6,17 @@ build = "build.rs" # <-- We added this and everything after!
 workspace = "../.."
 edition = "2021"
 
+[build-dependencies]
+regex-syntax = { workspace = true, features = ["unicode-perl"] }
+
+[dev-dependencies]
+regex = { workspace = true, features = ["unicode-perl"] }
+
 [build-dependencies.lalrpop]
 version = "0.19.10"
 path = "../../lalrpop"
 features = ["lexer"]
 
-[dependencies]
-regex = "1"
```

Comment:
This is part of document example code. Not using workspace will helps document readers to follow the code out of lalrpop workspace.

Reply:
deps updated not to use the workspace deps in doc examples 👍

Reply:
Makes me wonder why is regex required at that point in the tutorial?

IIUC, it only required in the `calculator` binary because it uses `\s` which requires the `unicode-perl` feature to be enabled.

Reply:
@youknowone I updated the doc (in the quick start guide) to raise attention on the possible need for the extra regex's unicode features.

I haven't figured yet what/when are the feature required on regex-syntax or regex.

Reply:
I'm honestly rather confused what all this breakage is about, so it's hard for me to have an informed opinion.

Reply:
It's a confusing situation, and I'm not sure I've seen a clear, concise write-up.  Let me try to provide one and see if it can clarify things for you.

When a user writes a regex in their .lalrpop file, lalrpop uses the regex-syntax crate to generate a representation of that regex (https://github.com/lalrpop/lalrpop/blob/master/lalrpop/src/lexer/re/mod.rs).  If that representation is generatable without unicode support, this always works.  However, if the regex requires unicode support, then regex-syntax *must* have a unicode feature enabled (I'm still a little vague on unicode vs unicode-perl.  I think just the unicode-perl feature will work in most situations).

A common situation here is a regex like `r"\s*"`, which matches both ASCII and unicode whitespace.

If regex-syntax is compiled without unicode support, it reports an error at runtime.

Lalrpop does not currently (on master) specify that the regex-syntax unicode feature is required.  However, due to feature unification it can be required in a number of situations.  Specifically, the regex crate depends on regex-syntax, and with its default features it includes regex-syntax unicode support.

The problem was first noticed when regex updated to version 1.8.  Version 1.8 of regex moved its regex-syntax dependency to 0.7, while lalrpop continued to depend on 0.6.  This means that anyone getting unicode support indirectly through regex (lalrpop and its users) *did* get unicode support on regex-syntax 0.7, but did *not* on regex-syntax 0.6, since regex no longer used 0.6.  Lalrpop however still used 0.6, and so built regex-syntax without unicode support, and therefore broke lalrpop users relying on it.  This was masked in lalrpop's testing by the fact that lalrpop CI runs with a Cargo.lock in place, so was testing against an older version of regex, which had the same regex-syntax dependency as lalrpop.

Fundamentally, lalrpop users shouldn't need to know about or worry about features on lalrpop dependencies.  If lalrpop updates to a new regex-syntax version, or switches to a different regex crate, that should be transparent to lalrpop users.  However, the current state (on master) is that whether regex-syntax is built with unicode support depends on a bunch of transitive dependency resolution in the specific crate.

The right answer here is definitely what is suggested in #789.  Lalrpop should expose a configuration knob to users who can opt in or out to unicode support with lalrpop directly, rather than adopting brittle workarounds by setting a feature on regex-syntax, which they may not depend on otherwise.  Lalrpop then owns the responsibility for turning around and setting regex-syntax's feature flags around unicode independently.

Separately, lalrpop should also improve future robustness by merging some variation on this PR here and some variation on #755, so that features are accurately and fully specified and tested.